airbyte-ci/connectors/base_images/README.md

```diff
@@ -6,21 +6,22 @@ Our connector build pipeline ([`airbyte-ci`](https://github.com/airbytehq/airbyt
 Our base images are declared in code, using the [Dagger Python SDK](https://dagger-io.readthedocs.io/en/sdk-python-v0.6.4/).
 
 - [Python base image code declaration](https://github.com/airbytehq/airbyte/blob/master/airbyte-ci/connectors/base_images/base_images/python/bases.py)
-- ~Java base image code declaration~ _TODO_
+- ~Java base image code declaration~ *TODO* 
 
-## Where are the Dockerfiles?
 
+## Where are the Dockerfiles?
 Our base images are not declared using Dockerfiles.
 They are declared in code using the [Dagger Python SDK](https://dagger-io.readthedocs.io/en/sdk-python-v0.6.4/).
 We prefer this approach because it allows us to interact with base images container as code: we can use python to declare the base images and use the full power of the language to build and test them.
 However, we do artificially generate Dockerfiles for debugging and documentation purposes.
 
-### Example for `airbyte/python-connector-base`:
 
+
+### Example for `airbyte/python-connector-base`:
 ```dockerfile
-FROM docker.io/python:3.9.18-slim-bookworm@sha256:44b7f161ed03f85e96d423b9916cdc8cb0509fb970fd643bdbc9896d49e1cad0
+FROM docker.io/python:3.9.19-slim-bookworm@sha256:b92e6f45b58d9cafacc38563e946f8d249d850db862cbbd8befcf7f49eef8209
```

Comment:
What are our options to change to `python:3.9.19-alpine`?

Debian moves a bit slower and is a bit bigger. Pretty big difference in vulnerability size

```
(⎈|docker-desktop:default)➜  Downloads grype python:3.9.19-slim-bookworm 
 ✔ Vulnerability DB                [no update available]  
 ✔ Pulled image                    
 ✔ Loaded image                                                                                                                                               python:3.9.19-slim-bookworm
 ✔ Parsed image                                                                                                   sha256:182cc99a2af6b2eb20b8da7e1a0d3661424b8fa7a0f0fbdac147e5e4ca8a3005
 ✔ Cataloged contents                                                                                                    34f145fff6b64d6199becdf2f8d67d2810b54df9d610a14f0c1989ab39d0c01e
   ├── ✔ Packages                        [118 packages]  
   ├── ✔ File digests                    [2,954 files]  
   ├── ✔ File metadata                   [2,954 locations]  
   └── ✔ Executables                     [809 executables]  
 ✔ Scanned for vulnerabilities     [118 vulnerability matches]  
   ├── by severity: 1 critical, 7 high, 25 medium, 3 low, 55 negligible (27 unknown)
   └── by status:   2 fixed, 116 not-fixed, 0 ignored 
```

vs

```
(⎈|docker-desktop:default)➜  Downloads grype python:3.9.19-alpine       
 ✔ Vulnerability DB                [no update available]  
 ✔ Pulled image                    
 ✔ Loaded image                                                                                                                                                      python:3.9.19-alpine
 ✔ Parsed image                                                                                                   sha256:a3003c79447cb9c6aa127216f7dd3fe0c746723003effce38674cff406bcee25
 ✔ Cataloged contents                                                                                                    d3a44294453f9c846b8a1d36120f54690ee52d77c376b8b7d8d4b6850408415e
   ├── ✔ Packages                        [47 packages]  
   ├── ✔ File digests                    [659 files]  
   ├── ✔ File metadata                   [659 locations]  
   └── ✔ Executables                     [141 executables]  
 ✔ Scanned for vulnerabilities     [11 vulnerability matches]  
   ├── by severity: 0 critical, 2 high, 8 medium, 0 low, 0 negligible (1 unknown)
   └── by status:   2 fixed, 9 not-fixed, 0 ignored 
```

Reply:
@wennergr we originally went for bookworm as some python package like Pandas require system dependencies which are not on alpine.
It's feasible to  add them to an alpine base but not straightforward...
https://gist.github.com/orenitamar/f29fb15db3b0d13178c1c4dd611adce2

So we picked bookworm for simplicity and compatibility...

I suggest to cut this new version (1.2.1) with bookworm, and then cut a new one (2.0.0) with alpine.

This would be a major version as it might not be usable by some connectors.

Our batch update connector flow will  lead to a best effort thing:
* All connectors incompatible with alpine will fail their CI build so they won't get updated.